### PR TITLE
Some bug fixes on BadgeBundle and ToolRepository in core bundle. 

### PR DESCRIPTION
**Pull request by ptsavdar**
Also added some enhancements in BadgeBundle.

### CODE REVIEW

**Comment by ptsavdar** [comment]
LGTM

File: plugin/badge/Controller/AdministrationController.php [hunk 1]
```diff
@@ -134,6 +134,8 @@ public function editAction(Request $request, Badge $badge, $page = 1)
                 return $this->redirect($this->generateUrl('icap_badge_admin_badges'));
             }
         } catch (\Exception $exception) {
+            var_dump($exception->getMessage());
+            die();
```
Comment (stefk) [inline plugin/badge/Controller/AdministrationController.php]: !

Comment (ptsavdar) [inline plugin/badge/Controller/AdministrationController.php]: Oops! Sorry! I'm removing it and re-commit my changes.

**Comment by ngodfraind** [comment]
LGTM

**Pull request merged**